judge.pl

```diff
@@ -873,10 +875,8 @@ sub main_loop {
     for (my $i = 0; ; $i++) {
         sleep $cfg->sleep_time;
         $log->rollover;
-        log_msg("pong\n") if $judge->update_state;
         log_msg("...\n") if $i % 5 == 0;
-        next if $judge->is_locked;
-        my ($r, $state) = prepare_problem();
+        my ($r, $state) = prepare_problem() or next;
```

Comment:
...instead, check `was_pinged` here.
You do not need `or next` because of `if $r` below